File: PseudoCoupHQ/Research/oracle/riscv/softfloat_slices/emulations/java/ui32_to_f64.java


public final class ui32_to_f64 {
    public static long ui32_to_f64_rm1(int v_arg) {
        final boolean v_i = (v_arg == 0x0);
        final boolean v__n4 = (v_i != true);
        final long v__m5 = Sf.sext1l(v__n4);
        final int v__k1 = Sf.ctlz32(v_arg);
        final int v_i2 = v__k1;
        final int v_i3 = (v_i2 + 0x15);
        final int v_i4 = (0x41d - v_i2);
        final long v_i5 = ((long)v_i4 & 0xffffffffL);
        final long v__sh2 = (v_i5 << (int)(0x34L & 63L));
        final long v_i8 = ((long)v_i3 & 0xffffffffL);
        final long v_i7 = ((long)v_arg & 0xffffffffL);
        final long v__sh3 = (v_i7 << (int)(v_i8 & 63L));
        final long v_i9 = v__sh3;
        final long v_i10 = (v__sh2 + v_i9);
        final long v__a6 = (v_i10 & v__m5);
        return v__a6;
    }
}

File: PseudoCoupHQ/Research/oracle/riscv/softfloat_slices/emulations/ruby/ui32_to_f64.rb
require_relative 'helpers'


def ui32_to_f64_rm1(v_arg)
  v_i = ((v_arg == 0x0) ? 1 : 0)
  v__n4 = (v_i ^ 0x1)
  v__m5 = (sf_sgn(v__n4, 0x1) & 0xffffffffffffffff)
  v__k1 = sf_ctlz(v_arg, 0x20)
  v_i2 = v__k1
  v_i3 = ((v_i2 + 0x15) & 0xffffffff)
  v_i4 = ((0x41d - v_i2) & 0xffffffff)
  v_i5 = (v_i4)
  v__sh2 = ((v_i5 << (0x34 & 0x3f)) & 0xffffffffffffffff)
  v_i8 = (v_i3)
  v_i7 = (v_arg)
  v__sh3 = ((v_i7 << (v_i8 & 0x3f)) & 0xffffffffffffffff)
  v_i9 = v__sh3
  v_i10 = ((v__sh2 + v_i9) & 0xffffffffffffffff)
  v__a6 = (v_i10 & v__m5)
  v__a6
end
